ingestion/src/metadata/ingestion/source/database/redshift/metadata.py

```diff
@@ -476,58 +375,24 @@ def get_partition_details(self) -> None:
         for row in results:
             self.partition_details[f"{row.schema}.{row.table}"] = row.diststyle
 
-    def get_tables_name_and_type(self) -> Optional[Iterable[Tuple[str, str]]]:
+    def query_table_names_and_types(
+        self, schema_name: str
+    ) -> Iterable[TableNameAndType]:
+        """
+        Handle custom table types
         """
-        Handle table and views.
 
-        Fetches them up using the context information and
-        the inspector set when preparing the db.
+        result = self.connection.execute(
+            sql.text(REDSHIFT_GET_ALL_RELATION_INFO),
```

Comment:
Same in redshift. No need to reimplement the full dialect patching the query. Just run a simplified version returning name and relkind for local and external tables.

We keep the default implementation for views as that does not change